trpcultivate_phenotypes/src/Controller/PhenoExperimentTraitHandler.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+namespace Drupal\trpcultivate_phenotypes\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Database\Connection;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * Phenotypes experiment configuration trait operations.
+ */
+class PhenoExperimentTraitHandler extends ControllerBase {
+
+  /**
+   * Drupal database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected Connection $db;
+
+  /**
+   * Request array.
+   *
+   * @var array
+   */
+  private array $request;
+
+  /**
+   * The table name.
+   *
+   * @var string
+   */
+  private const TABLE_NAME = 'trpcultivate_phenocombo';
+
+  /**
+   * Expected parameter per action.
+   *
+   * @var array
+   */
+  private const ACTION_PARAMS = [
+    'assign' => [
+      'label',
+      'required',
+      'combo',
+      'project',
+      'genus',
+      'user',
+    ],
+    'remove' => [
+      'combo_id',
+    ],
+  ];
+
+  /**
+   * Good response code.
+   *
+   * @var int
+   */
+  private const GOOD_REQUEST = 200;
+
+  /**
+   * Bad response code.
+   *
+   * @var int
+   */
+  private const BAD_REQUEST = 400;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Database\Connection $db
+   *   Drupal database connection.
+   */
+  public function __construct(Connection $db) {
+    $this->db = $db;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('database'),
+    );
+  }
+
+  /**
+   * Handle AJAX request to assign/remove trait to/from experiment.
+   *
+   * @param Symfony\Component\HttpFoundation\Request $request
+   *   Request.
+   * @param string $action
+   *   The requested action - assign or remove a trait.
+   *
+   * @return Symfony\Component\HttpFoundation\JsonResponse
+   *   JSON response array message and status code.
+   */
+  public function handleAction(Request $request, string $action) {
```

Comment:
I think these docs could use some expanded to make how this is happening more clear. Maybe something like:


```suggestion
   * Handle AJAX request to assign/remove trait to/from experiment.
   *
   * The "trait" and "experiment" this action will apply to is encoded in the
   * request object and will be saved to self::request.
   *
   * @param Symfony\Component\HttpFoundation\Request $request
   *   Request.
   * @param string $action
   *   The requested action to be peformed for a trait. Supported actions are:
   *   - assign: assigns the trait to the experiment.
   *   - remove: removes the trait from the experiment.
   *
   * @return Symfony\Component\HttpFoundation\JsonResponse
   *   JSON response array message and status code.
   */
  public function handleAction(Request $request, string $action) {
```